File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/java/src/main/java/de/sots/cellarsandcentaurs/domain/model/ArmorClass.java
package de.sots.cellarsandcentaurs.domain.model;

import de.sots.cellarsandcentaurs.application.CreatureUtil;

public class ArmorClass {
    private String description;
    private int base;
    private int bonus;
    private int total;

    public ArmorClass(int base, int bonus) {
        this(base, bonus, CreatureUtil.STANDARD_ARMOR_CLASS_DESCRIPTION);
    }

    public ArmorClass(int base, int bonus, String description) {
        this.description = description;
        this.base = base;
        this.bonus = bonus;
        this.total = base + bonus;
    }

    public int getBase() {
        return base;
    }

    public void setBase(int base) {
        this.base = base;
    }

    public int getBonus() {
        return bonus;
    }

    public void setBonus(int bonus) {
        this.bonus = bonus;
    }

    public int getTotal() {
        return total;
    }

    public void setTotal(int total) {
        this.total = total;
    }
}

File: analysis/analysers/parsers/DependencyParser/src/test/resources/analysis/contract/examples/cpp/src/de/sots/cellarsandcentaurs/domain/model/ArmorClass.cpp
#include "ArmorClass.h"
#include "../../application/CreatureUtil.h"

namespace de::sots::cellarsandcentaurs::domain::model {

ArmorClass::ArmorClass(int base, int bonus)
    : ArmorClass(base, bonus, de::sots::cellarsandcentaurs::application::CreatureUtil::STANDARD_ARMOR_CLASS_DESCRIPTION)
{}

ArmorClass::ArmorClass(int base, int bonus, const std::string& description)
    : description_{description}, base_{base}, bonus_{bonus}, total_{base+bonus}
{}

int ArmorClass::getBase() const noexcept { return base_; }
void ArmorClass::setBase(int base) noexcept { base_ = base; total_ = base_ + bonus_; }

int ArmorClass::getBonus() const noexcept { return bonus_; }
void ArmorClass::setBonus(int bonus) noexcept { bonus_ = bonus; total_ = base_ + bonus_; }

int ArmorClass::getTotal() const noexcept { return total_; }
void ArmorClass::setTotal(int total) noexcept { total_ = total; }

const std::string& ArmorClass::getDescription() const noexcept { return description_; }
void ArmorClass::setDescription(const std::string& description) { description_ = description; }

}
